workflow/controller/dag.go

```diff
@@ -51,6 +51,15 @@ func (d *dagContext) taskNodeName(taskName string) string {
 	return fmt.Sprintf("%s.%s", d.boundaryName, taskName)
 }
 
+func (d *dagContext) getTaskFromNode(node *wfv1.NodeStatus) *wfv1.DAGTask {
+	for _, task := range d.tasks {
+		if node.Name == d.taskNodeName(task.Name) {
+			return &task
+		}
+	}
+	panic("task from node " + node.Name + " does not exist")
```

Comment:
Panicking will crash the controller. Can this ever happen in the real world?

Reply:
Nope, should be unreachable. A similar function is above this